src/main/java/org/logstash/filters/DateFilter.java

```diff
@@ -77,22 +77,24 @@ public void acceptFilterConfig(String format, String locale, String timezone) {
       Event event = rubyEvent.getEvent();
       // XXX: Check for cast failures
 
-      switch (executeParsers(event)) {
-        case FIELD_VALUE_IS_NULL_OR_FIELD_NOT_PRESENT:
-          continue;
-        case SUCCESS:
-          if (successHandler != null) {
-            successHandler.handle(rubyEvent);
-          }
-          break;
-        case FAIL: // fall through
-        default:
-          for (String t : tagOnFailure) {
-            event.tag(t);
-          }
-          if (failureHandler != null) {
-            failureHandler.handle(rubyEvent);
-          }
+      if (! event.isCancelled()) {
```

Comment:
+1 on the strategy, I just suggest not adding an extra level of indentation for the whole block by just adding:

```java
if event.isCancelled()
  continue;
```

leaving everything else the same.